### ISSUE TITLE
XML-RPC API documentation on the Wiki is not up to date for version 3.3.0.

### ISSUE THREAD

**Issue by loudonlune** [cobbler/cobbler#2896]
# Issue

The documentation on the wiki for the XML-RPC API is out of date for 3.3.0 after the web interface was removed. 

## Description
The "modify_interfaces" setter for system objects no longer functions as shown in the section headed with "An example of how to add a new host". System objects in 3.3.0 appear to have a new setter for interfaces that I had to find by sifting through your source code. I don't know what other silent changes to the API have occurred, but please update the wiki to reflect these changes.

**Comment by SchoolGuy** [comment]
I am deeply sorry that we overlooked this. Thanks for reporting it. The manpower is low but we'll do our best!

**Comment by SchoolGuy** [comment]
I found the section in the Wiki. The page is referring to `xmlrpclib` which is Python 2. I will thus overhaul the page in a  greater manner.

**Comment by SchoolGuy** [comment]
This bug is a duplicate of #2846. This is re-enabled as of 3.3.1. The rest of the wiki page is still getting an update.

```
Show Cobbler version
Cobbler 3.3.1
  source: f5b0599a, Fri Feb 18 12:55:15 2022 +0100
  build time: Mon May  1 09:49:35 2023
60a8dfc9857a:/code # mkdir -p /opt/stuff
60a8dfc9857a:/code # touch /opt/stuff/vmlinuz
60a8dfc9857a:/code # touch /opt/stuff/initrd.img
60a8dfc9857a:/code # python3
Python 3.6.13 (default, Mar 10 2021, 18:30:35) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import xmlrpc.client
>>> remote =  xmlrpc.client.Server("http://127.0.0.1/cobbler_api")
>>> token = remote.login("cobbler","cobbler")
>>> distro_id = remote.new_distro(token)
>>> remote.modify_distro(distro_id, 'name',   'example-distro',token)
True
>>> remote.modify_distro(distro_id, 'kernel', '/opt/stuff/vmlinuz',token)
True
>>> remote.modify_distro(distro_id, 'initrd', '/opt/stuff/initrd.img',token)
True
>>> remote.save_distro(distro_id,token)
True
>>> profile_id = remote.new_profile(token)
>>> remote.modify_profile(profile_id, "name", "example-profile", token)
True
>>> remote.modify_profile(profile_id, "distro", "example-distro", token)
True
>>> remote.save_profile(profile_id,token)
True
>>> system_id = remote.new_system(token)
>>> remote.modify_system(system_id,"name","hostname",token)
True
>>> remote.modify_system(system_id,"hostname","hostname.example.com",token)
True
>>> remote.modify_system(system_id,"profile","example-profile",token)
True
>>> remote.save_system(system_id, token)
True
>>> remote.modify_system(system_id, "modify_interface", {"macaddress-eth0": "01:02:03:04:05:06", "ipaddress-eth0": "192.168.0.1", "dnsname-eth0": "hostname.example.com"}, token )
True
>>> remote.save_system(system_id, token)
True
```

**Comment by SchoolGuy** [comment]
Fixed by #2921
